src/main/scala/dataclass/Macros.scala

```diff
@@ -127,12 +127,12 @@ private[dataclass] class Macros(val c: Context) extends ImplTransformers {
                               q"${p.name}=_root_.scala.Some(${p.name})"
                             )
                           )
-                        q"def $withDefIdent(${p.name}: $tpe0) = new $tpname[..$tparamsRef](...$namedArgs0)"
+                        q"def $withDefIdent(${p.name}: $tpe0) = ${tpname.toTermName}[..$tparamsRef](...$namedArgs0)"
```

Comment:
I dont have much experience with scala meta, so I am not sure if this is the best way to call apply method, but seems to work...

Reply:
I'm not really sure either, but if the CI is green, that must fine…